iceoryx_hoofs/include/iceoryx_hoofs/internal/containers/uninitialized_array.inl

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_INL
+#define IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_INL
+
+#include "iceoryx_hoofs/containers/uninitialized_array.hpp"
+
+namespace iox
+{
+namespace containers
+{
+template <typename ElementType, uint64_t Capacity, template <typename, uint64_t> class Buffer>
+inline constexpr ElementType&
+UninitializedArray<ElementType, Capacity, Buffer>::operator[](const uint64_t index) noexcept
+{
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : type safety ensured by template parameter
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    return *reinterpret_cast<ElementType*>(&m_buffer.value[index]);
```

Comment:
I would be alright to not use the `const_cast` approach. Code duplication is for me not a matter of reducing lines of code but to put the logic at only one place.
If the logic has to be adjusted and there are multiple places it can quickly happen that one code snippet is fixed while the other one still contains the old behavior.